starlite/testing/sync_test_client/transport.py

```diff
@@ -0,0 +1,82 @@
+from io import BytesIO
+from typing import TYPE_CHECKING, Any, Dict, cast
+
+from anyio import Event
+from httpx import BaseTransport, ByteStream, Request, Response
+
+from starlite.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
+from starlite.testing.base.transport_base import (
+    BaseClientTransport,
+    ConnectionUpgradeException,
+    SendReceiveContext,
+)
+from starlite.testing.sync_test_client.websocket_test_session import (
+    WebSocketTestSession,
+)
+
+if TYPE_CHECKING:
+    from starlite.testing.sync_test_client.client import TestClient
+    from starlite.types import WebSocketScope
+
+
+class TestClientTransport(BaseTransport, BaseClientTransport):
+    def __init__(
+        self,
+        client: "TestClient",
+        raise_server_exceptions: bool = True,
+        root_path: str = "",
+    ) -> None:
+        BaseClientTransport.__init__(
+            self, client=client, raise_server_exceptions=raise_server_exceptions, root_path=root_path
+        )
+
+    def handle_request(self, request: "Request") -> "Response":
```

Comment:
I am also pretty sure the antire transport is redundant. 

You can simple have a handle_request method, and then a `handle_async_request` method that calls the other method and wraps it in async logic.

Reply:
@Goldziher I am not sure I completely understand this part, but if you could review the changes and see if I understood
right